pkg/process/net/resolver/resolver.go

```diff
@@ -12,20 +12,70 @@ import (
 	"sync"
 	"time"
 
+	"github.com/benbjohnson/clock"
+
 	model "github.com/DataDog/agent-payload/v5/process"
 
 	procutil "github.com/DataDog/datadog-agent/pkg/process/util"
+	proccontainers "github.com/DataDog/datadog-agent/pkg/process/util/containers"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
 const defaultTTL = 10 * time.Second
 
 // LocalResolver is responsible resolving the raddr of connections when they are local containers
 type LocalResolver struct {
-	mux         sync.RWMutex
-	addrToCtrID map[model.ContainerAddr]string
-	ctrForPid   map[int]string
-	updated     time.Time
+	mux                sync.RWMutex
+	addrToCtrID        map[model.ContainerAddr]string
+	ctrForPid          map[int]string
+	updated            time.Time
+	lastContainerRates map[string]*proccontainers.ContainerRateMetrics
+	Clock              clock.Clock
+	ContainerProvider  proccontainers.ContainerProvider
+	done               chan bool
+}
+
+func NewLocalResolver(containerProvider proccontainers.ContainerProvider, clock clock.Clock) *LocalResolver {
+	return &LocalResolver{
+		ContainerProvider: containerProvider,
+		Clock:             clock,
+		done:              make(chan bool),
+	}
+}
+
+func (l *LocalResolver) Run() {
+	ticker := l.Clock.Ticker(10 * time.Second)
+	go l.pullContainers(ticker)
+}
+
+func (l *LocalResolver) Stop() {
+	l.done <- true
+}
+
+func (l *LocalResolver) pullContainers(ticker *clock.Ticker) {
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			var containers []*model.Container
+			var pidToCid map[int]string
+			var lastContainerRates map[string]*proccontainers.ContainerRateMetrics
+
+			cacheValidityNoRT := 2 * time.Second
+			containers, lastContainerRates, pidToCid, err := l.ContainerProvider.GetContainers(cacheValidityNoRT, l.lastContainerRates)
+			if err == nil {
+				l.lastContainerRates = lastContainerRates
+			} else {
+				log.Debugf("Unable to gather stats for containers, err: %v", err)
```

Comment:
Shouldn't this be a non-debug error log?

Reply:
This was also a debug level log in the container and process check, so I kept as such here. Not sure why its debug in the first place. Do you think it should be at warn level?

Reply:
Yes I think that would be more appropriate here.

Reply:
Done